core/blockchain.go

```diff
@@ -1411,6 +1422,8 @@ func (bc *BlockChain) writeBlockWithState(block *types.Block, receipts []*types.
 	if err != nil {
 		return err
 	}
+	// Azamat: It seems like a good place to send state changes, because they are committed to the DB
+	bc.stateChangeFeed.Send(extractStateChanges(state))
```

Comment:
Maybe send also, block number ? Is it already in state ?

Reply:
We for sure need to send `blockNumber`, but I didn't check, where to get from. I thought it should be quite trivial :grin: 
If not in state, we can take it from block or just send the extracted from block